### PR TITLE
Add zsh completion for main duperemove command.

### PR DESCRIPTION
**Pull request by VorpalBlade**
This PR adds zsh compltion for the main duperemove command. I did not create zsh completion for the other commands as after reading the man pages they appeared to be more of debugging commands and not worth the effort.

Unfortunately I don't have the skill to write an equivalent bash completion file.

### CODE REVIEW

**Comment by VorpalBlade** [comment]
Any reason this PR can't be merged? This has been here for quite a while. And I see the repository has activity again. @JackSlateur

**Comment by JackSlateur** [comment]
Hello @VorpalBlade 

Thank you for your contribution
If you could update the changeset to include the modifications I reviewed, that would be awesome

**Comment by VorpalBlade** [comment]
> Thank you for your contribution If you could update the changeset to include the modifications I reviewed, that would be awesome

I don't see any review comments. Did you remember to mark it as finished/ready in Github so it is shared with others?

**Comment by JackSlateur** [comment]
> I don't see any review comments. Did you remember to mark it as finished/ready in Github so it is shared with others?

Ha yes, that would be better

File: completion/zsh/_duperemove [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#compdef duperemove
+
+_duperemove_options()
+{
+	declare -a dedupe_options
+	dedupe_options=(
```
[the remark this answers is not in the document]
Comment (VorpalBlade) [reply completion/zsh/_duperemove:6]: Fixed, also according to man page same is now default, while before nosame was default

File: completion/zsh/_duperemove [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		'(nopartial)partial[compare portions of extents (default)]'
+		'(same)nosame[do not allow dedupe of extents within the same file (default)]'
+		'(nosame)same[allow dedupe of extents within the same file]'
+	)
+	_values -s , option $dedupe_options
+}
+
+_duperemove() {
+	declare -a args
+	args=(
```
[the remark this answers is not in the document]
Comment (VorpalBlade) [reply completion/zsh/_duperemove:19]: Fixed with one caveat. zsh will assume that all arguments after -R are arguments to -R. I don't know if that can be solved. The syntax is supposedly `optspec:*pattern:message:action` where the pattern describes when to stop, but it seems to assume that pattern describes a specific end flag (e.g. `-R file1 file2 END_SYMBOL -other-flag`) which I don't think is correct.

**Comment by VorpalBlade** [comment]
Fixed (with the one noted caveat in the response to your comments for -R)

**Comment by JackSlateur** [comment]
Perfect
One last issue: when completion is enabled, auto-completion of files no longer works

It may miss a line into the args array to define the file list: when I write `duperemove <someoptions> <tab> <tab>`, it should be completed by the files in current working directory

This is not clear .. I do not know the words to say this

**Comment by JackSlateur** [comment]
Awesome

Thank you for your contribution !

**Pull request merged**